spring-cloud-aws-core/src/main/java/io/awspring/cloud/core/config/AmazonWebserviceClientFactoryBean.java

```diff
@@ -112,18 +112,10 @@ protected T createInstance() throws Exception {
 
 		if (this.customEndpoint != null) {
 			builder.withEndpointConfiguration(
-					new AwsClientBuilder.EndpointConfiguration(this.customEndpoint.toString(), null));
+					new AwsClientBuilder.EndpointConfiguration(this.customEndpoint.toString(), findRegion()));
```

Comment:
Isn't this setting the "signing region" while what you want is to set the "region". So maybe this line should be unaffected, and the `else` block below should just always happen, and not just if there isn't a custom endpoint.

Reply:
Only one of `endpointConfiguration` and `region` may be set, so block below cannot always happen.

Reply:
Do you not agree that the second argument to the `EndpointConfiguration` constructor is `signingRegion`? If so, isn't that something different than "region", which is what we're getting here from `findRegion`?

Reply:
I do agree that it is a signing region. I mean we cannot do `setEndpoint` and `setRegion` at once.

I am not sure if there is any drawback of always passing a signing region, AWS SDK source code says that in most cases clients should just pass null.

Reply:
I don't feel I'm enough of an expert to make a definitive statement here, but I do feel strong that my need is to set the region, not the signing region. (I came here because I recognize the problem as the same that I'm having, and setting the region seems the solution.)